### PR TITLE
fix: Forbid SSO users from logging in using passwordless

### PR DESCRIPTION
**Pull request by codingllama**
It is possible for an SSO user to register a passkey and skip SSO authentication by doing a local/passwordless login. This PR fixes that by checking the user "type" on passwordless logins.

Note that, for passwordless, the user is only confirmed after the challenge is solved, so the error has to happen in the "finish" step of the ceremony. Similarly the Web UI can't do much to identify the user or forbid passwordless before that.

Changelog: Fix user SSO bypass by performing a local passwordless login

### CODE REVIEW

File: lib/auth/helpers_mfa.go [hunk 1]
```diff
@@ -221,8 +221,7 @@ func (d *TestDevice) solveRegisterWebauthn(c *proto.MFARegisterChallenge) (*prot
 	d.Key.PreferRPID = true
 
 	if d.passwordless {
-		d.Key.AllowResidentKey = true
-		d.Key.SetUV = true
+		d.Key.SetPasswordless()
```
Comment (codingllama) [inline lib/auth/helpers_mfa.go:224]: This here was missing a flag, plus at some point I added SetPasswordless() but probably missed this line.

File: lib/auth/auth_login_test.go [hunk 2]
```diff
@@ -1083,6 +1079,94 @@ func TestServer_Authenticate_passwordless(t *testing.T) {
 	}
 }
 
+// TestServer_Authenticate_passwordlessSSO tests a scenario where an SSO user
+// bypasses SSO by using a passwordless login.
+func TestServer_Authenticate_passwordlessSSO(t *testing.T) {
```
Comment (zmb3) [inline lib/auth/auth_login_test.go:1084]: You're free to continue to disagree with me, but I will continue to express my distaste for mixing upper case, lower case, and underscores in a function name.

```suggestion
func TestPasswordlessProhibitedForSSO(t *testing.T) {
```

**Review by zmb3 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

File: lib/auth/auth_login_test.go (hunk 2 above)
Comment (codingllama) [reply lib/auth/auth_login_test.go]: Done, here and at lib/web.

**Comment by codingllama** [comment]
Thanks for the quick reviews! I'm trying to land a few #41023 backports before I merge this one, to save myself some trouble. I should be able to proceed here soon.

**Comment by codingllama** [comment]
Hey @gregnetau, this change targets only passwordless logins, other kinds of MFA are unaffected. This includes MFA checks with a resident credential.

If you are having problems, please raise an issue or a support ticket with details of your problem and repro steps and we'll take a look at it.

**Pull request merged**